internal/mode/static/state/graph/policies.go

```diff
@@ -83,11 +84,54 @@ func (g *Graph) attachPolicies(ctlrName string) {
 				}
 
 				attachPolicyToRoute(policy, route, ctlrName)
+			case kinds.Service:
+				svc, exists := g.ReferencedServices[ref.Nsname]
+				if !exists {
+					continue
+				}
+
+				attachPolicyToService(policy, svc, g.Gateway, ctlrName)
 			}
 		}
 	}
 }
 
+func attachPolicyToService(
+	policy *Policy,
+	svc *ReferencedService,
+	gw *Gateway,
+	ctlrName string,
+) {
+	var ancestor *PolicyAncestor
+
+	for _, parentGw := range svc.ParentGateways {
+		if parentGw != client.ObjectKeyFromObject(gw.Source) {
+			continue
+		}
+		// Once we support multiple Gateways, this will turn into a list.
```

Comment:
> Is this because a Route could be attached to multiple Gateways? And this policy could be attached to a Service in that Route?

Yes. 

> Would we then need to pass in multiple gateways to this function?

I think so